### DISCUSSION TITLE
I used API to create product and got 502 Bad Getaway error.

### DISCUSSION THREAD

**Discussion by raj0319** [Sellbrite/Sellbrite-API#18] [Q&A]
my code in like below,

// Sellbrite API endpoint for creating a product
$endpoint = "https://api.sellbrite.com/v1/products";

// API credentials
$api_key = '[REDACTED]';
$account_id = '[REDACTED]';

// Data for creating the product
$product_data = array(
    "sku" => "PRODUCT_SKU",
    "name" => "Product Name",
    "description" => "Product Description",
    "inventory_sku" => "INVENTORY_SKU",
    "inventory_quantity" => 10,
    "cost" => 25.99,
    // Add more product details as needed
);

// Initialize cURL session
$ch = curl_init();

// Set cURL options
curl_setopt($ch, CURLOPT_URL, $endpoint);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($product_data));
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    "Content-Type: application/json",
    "Authorization=Basic $account_id:$api_key"
));

// Execute cURL request and get response
$response = curl_exec($ch);

// Check for cURL errors
if(curl_errno($ch)){
    echo 'Curl error: ' . curl_error($ch);
}

// Close cURL session
curl_close($ch);

// Decode and display the response
$response_data = json_decode($response, true);
if (isset($response_data['product'])) {
    echo "Product created successfully. Product ID: " . $response_data['product']['id'];
} else {
    echo "Error creating product. Response: " . print_r($response_data, true);
}
exit;

**Comment by rflores1-godaddy** [comment]
Hello, thank you for reaching out!

In this situation, the 502 error indicates that the server ran into an unexpected error and was unable to process the request. The payload appears to be valid, so you should be able to successfully re-run the script after a few hours or up to a day has passed. Please let us know if the error persists and we will provide further assistance.

Additionally, I strongly recommend against including sensitive information such as your API key in public posts. I have edited the post to exclude this information, but to maintain your account's security you need to rotate your API key as soon as possible.
